### ISSUE TITLE
Order of the "Other ID" code table

### ISSUE THREAD

**Issue by Jegelewicz** [ArctosDB/arctos#1902]
Does anyone have a problem with a change in order to the "Other ID Type" code table? Currently AF and ALA are listed as 1 and 2. I love you guys, but I think we should have the most universal types at the top. I propose that we let all other types that refer to other museum collections and entities sort alphabetically AFTER those types that are more generic.

So instead of the order we currently have. These would appear at the top:

barcode
--
collector   number
field   number
institutional catalog number
Locality ID
original   identifier
Original Image Number
preparator   number
published   identifier
second   collector number
secondary   identifier
toe   clip number
trap number
Trapline ID
vessel   identifier

Also, can we talk about making them all lower case or proper case for consistency?

**Comment by campmlc** [comment]
I'm biased here, because we use NK daily and I know the same is true for AF
at Alaska. Whereas vessel identifier is hardly used. What options do we
have for filtering these dropdowns to make them more user friendly? Can we
type "NK" without having to scroll through the list? Searching dropdowns
this long is tedious and painful. What can we do to make this easier and
potentially more specific to each collections needs? This should apply to
all dropdowns in Arctos, including the list of collections.

**Comment by dustymc** [comment]
>What options do we have for filtering these dropdowns to make them more user friendly?

I think anything you do to make them more "friendly" to one user is probably going to make them more evil to another.

I would personally start by getting rid of about half the stuff in there.

```
UAM@ARCTOS> select count(*) from CTCOLL_OTHER_ID_TYPE;

  COUNT(*)
----------
       400

1 row selected.

Elapsed: 00:00:00.02
UAM@ARCTOS> select count(*) from CTCOLL_OTHER_ID_TYPE where base_url is not null;

  COUNT(*)
----------
       182

1 row selected.
```

Is anyone searching on "LFL: CSU Larval Fish Lab" (or any of the seldom-used types which don't have a URL)? Would having that as type="original identifier"  + value="LFL: CSU Larval Fish Lab Number Twelve" somehow reduce the quality of the data? If the answer is "not really" then why do we have it? http://handbook.arctosdb.org/documentation/other-identifying-numbers.html#general-guidelines

There are also a few things that look like "random things some collector-like entity scribbled on something" to me - perhaps some of them can be merged.

```
collector number
original identifier
preparator number
secondary identifier
institutional catalog number
vessel identifier
Trapline ID
second collector number
trap number
field number
published identifier
MVZ alternate accession number
```



SpecimenSearch uses a dropdown replacement library because someone asked for it (probably to make it "more friendly!"). My preference is for "normal" HTML objects unless there's a compelling reason to do otherwise, but changing that needs an Issue. In addition to the normal HTML SELECT object, there are a ton of jquery libraries with demos - just google "jquery select replacement." (And we're not necessarily restricted to SELECT or jQuery.)

Here's the data by usage, which isn't necessarily a perfect reflection of how users access the list - I assume most users have a "custom ID" and would like the list sorted by what they use second most commonly.

```
UAM@ARCTOS> select OTHER_ID_TYPE || ' @ ' || count(*) from coll_obj_other_id_num group by OTHER_ID_TYPE order by count(*) desc;

OTHER_ID_TYPE||'@'||COUNT(*)
------------------------------------------------------------------------------------------------------------------------
collector number @ 1206184
original identifier @ 876927
AHRS (Alaska Heritage Resources Survey) @ 625854
Locality ID @ 478548
barcode @ 237338
ALAAC @ 236186
preparator number @ 215071
NK @ 205247
secondary identifier @ 84807
AF @ 78475
MLZ: Moore Laboratory of Zoology @ 61179
institutional catalog number @ 55132
NEON sample ID @ 54231
U. S. National Park Service catalog @ 53451
USGS: HUC 8-digit @ 49931
USGS: U.S. Geological Survey @ 46331
U. S. National Park Service accession @ 41463
DGR: Division of Genomic Resources (MSB) @ 37928
UIMNH: University of Illinois Museum of Natural History @ 32780
MSB:Para @ 27069
ISC: Ada Hayden Herbarium, Iowa State University @ 26645
GenBank @ 26458
UUID @ 24688
NEON: National Ecological Observatory Network @ 20551
MSB:Mamm @ 16004
MSB:Host @ 15548
UAM:Mamm @ 13326
MVZ salamander tissue collection @ 12262
MVZ frozen tissue collection @ 11744
DZTB: Denver Zoology Tissue Bird @ 10381
IF: Idaho Frozen Tissue Collection @ 8913
DZTM: Denver Zoology Tissue Mammal @ 8317
UAM:Ento @ 7735
MBM-UNLV: Marjorie Barrick Museum, University of Nevada Las Vegas @ 7619
UAM:Moll: University of Alaska Museum Mollusc @ 7158
BoLD barcode ID @ 6306
vessel identifier @ 6266
NMMNHS: New Mexico Museum of Natural History and Science @ 5805
GWRC: Greenwood Wildlife Rehabilitation Center @ 5714
BSFC: Biological Survey Fort Collins @ 4682
Trapline ID @ 4341
second collector number @ 3965
MVZ anatomical collection @ 3336
USFS: US Forest Service @ 2918
BSI: Biotic Survey and Inventory @ 2876
MVZ:Mamm @ 2874
ADFG: Alaska Department of Fish and Game @ 2187
CHAS:Inv @ 2096
KNWR:Ento @ 2013
UCM Osteology @ 1619
MVZ Prep Lab Catalog @ 1516
Kay P. Yanev Frozen Catalog @ 1472
USAC: Museo de Historia Natural, Universidad de San Carlos de Guatemala @ 1453
UAM:Herb @ 1253
MVZ:Herp @ 1244
MSB:Fish @ 1199
CWRF: Colorado Wild Rabbit Foundation @ 1164
Mexican wolf studbook number @ 1142
LWH: Lindsay Wildlife Hospital @ 1127
DMNS: Denver Museum of Nature and Science @ 1107
MVZ:Bird @ 1029
Instituto Nacional Pesquisas da Amazônia @ 975
USNPC: United States National Parasite Collection @ 912
DMNS:Mamm @ 879
RMRP: Rocky Mountain Raptor Program @ 832
CHAS:Bird @ 784
AMNH: American Museum of Natural History @ 763
NMSU: New Mexico State University @ 755
ALMNH PaleoBotany Collection @ 744
Museu Paraense Emílio Goeldi, Belém, Brazil @ 688
Alpha Helix Expedition @ 676
TTU: Texas Tech University @ 671
UNAM: Universidad Nacional Autonoma de Mexico @ 666
Milton Hildebrand collection @ 661
LVNP: Lassen Volcanic National Park @ 617
Museu de Zoologia, Universidade de São Paulo, Brazil @ 592
UCM Synoptic Teaching Collection @ 576
LACM: Los Angeles County Museum @ 551
UAM:Bryo: University of Alaska Museum Bryozoan @ 524
USU: Utah State University @ 522
DMNS:Para @ 505
USNM: National Museum of Natural History, Smithsonian Institution @ 424
LSU: Louisiana State University Museum of Natural Science @ 412
ZIN: Zoological Institute, Russian Academy of Science @ 395
UAM:Crus: University of Alaska Museum Marine Arthropod @ 375
USFWS: U.S. Fish and Wildlife Service @ 374
CMNH: Carnegie Museum of Natural History @ 369
MSB:Bird @ 364
NBSB: National Biomonitoring Specimen Bank @ 338
KUMNH: University of Kansas Museum of Natural History @ 334
UWYMV Frozen Tissue Collection @ 322
UTEP:Herp @ 321
trap number @ 313
CHAS:Mamm @ 304
UMMZ: University of Michigan Museum of Zoology @ 278
MVZObs:Bird @ 254
CAS: California Academy of Sciences, San Francisco @ 248
RSA-POM: Herbarium of Rancho Santa Ana Botanic Garden and Pomona College @ 244
AES: Agricultural Experimental Station Herbarium @ 223
field number @ 222
BioSample @ 203
AMMTAP: Alaska Marine Mammal Tissue Archival Project @ 196
UAM: University of Alaska Museum @ 186
MVZ:Egg @ 181
UAMObs:Ento @ 179
OWMZ: Ohio Wesleyan Museum of Zoology @ 174
NCSM: North Carolina State Museum @ 173
Museo Nacional de Ciencias Naturales, Madrid @ 170
MSB: Museum of Southwestern Biology @ 166
published identifier @ 161
PRBO: Point Reyes Bird Observatory @ 147
UCM:Mamm @ 146
CHAS:Egg @ 141
Museu Nacional do Rio de Janeiro @ 140
AMNH:Mamm @ 135
OMNH: Oklahoma Museum of Natural History @ 127
iNaturalist.org @ 119
DMNS:Bird @ 118
ACVC: Alameda County Vector Control @ 113
UCM Public Education and Exhibition Collection @ 110
TEX-LL: University of Texas at Austin, Plant Resources Center @ 109
Endangered Species Recovery Program, CSU Stanislaus @ 107
USNPC:Para @ 101
Florida Museum of Natural History, University of Florida @ 95
Howard Manter Parasitology Laboratory, University of Nebraska, Lincoln @ 94
UTEP:ES @ 94
KNWR:Herb @ 92
SDNHM: San Diego Natural History Museum @ 85
UCM:Bird @ 85
National Gallery of Art @ 82
Centennial Museum and Desert Gardens @ 78
Museo de Historia Natural de la Univ. Nacional Mayor de San Marcos, Lima @ 78
Kit Fox Recovery Project, ESRP, CSU Stanislaus @ 77
ASK @ 71
UTEP:HerpOS @ 71
CA Dept. Fish and Game, Ecoregion Biodiversity Monitoring-Northern Region @ 70
DGR:Fish @ 68
DGR:Mamm @ 66
KWP:Ento @ 64
University of Mississippi @ 58
UAM:EH @ 55
Hoopa Study Area Barred Owl Identifier @ 54
NMU:Mamm @ 53
TCWC: Texas Cooperative Wildlife Collection, Texas A & M University @ 41
Black Mountain Experimental Forest @ 40
UAM:Art @ 40
Kunming Institute, Yunnan Prov., China @ 39
U. S. Environmental Protection Agency @ 37
The Living Desert @ 36
KSU: Kansas State University @ 36
MCZ: Museum of Comparative Zoology, Harvard University @ 35
CHAS:Ento @ 35
UTEP:Mamm @ 34
DMNS:Egg @ 34
FMNH: Field Museum of Natural History @ 33
MVZ alternate accession number @ 31
Dennison Manufacturing Company @ 31
UAM:Fish @ 29
NCBI Biosample @ 28
Houston Zoo @ 27
The SPCA for Monterey County @ 24
Museum Zoologicum Bogoriense, Bogor, Indonesia:  @ 24
MCZ:Orn - Museum of Comparative Zoology Ornithology Collection @ 23
SBMNH: Santa Barbara Museum of Natural History @ 22
Universidad de la Republica, Montevideo @ 22
UCMP: University of California Museum of Paleontology @ 22
UCMP Locality ID @ 21
BugGuide.net @ 20
Fort Bliss Curatorial Facility @ 19
DNHM: Delaware Natural History Museum @ 19
California State Polytechnic University, Pomona @ 18
DMNS:Herp @ 17
NMFS: United States National Marine Fisheries Service @ 17
BMNH: Bell Museum of Natural History @ 17
UCSB: University of California Santa Barbara @ 17
OSUM: Oregon State University Museum @ 16
UTEP:Bird @ 16
MVZ:Img @ 16
PRC:TEX: University of Texas Herbarium @ 15
RBCM: Royal British Columbia Museum, Victoria, BC, Canada. @ 15
UWBM:Herp @ 15
CMNH:Bird: Cleveland Museum of Natural History Bird catalog @ 15
BYU: Monte L. Bean Life Science Museum, Brigham Young University @ 15
UCD: University of California, Davis @ 15
HWML:Para @ 14
MSB:Arth @ 14
Universidad Catolica de Quito @ 14
EA: National Museums of Kenya @ 13
NCBI Sequence Read Archive Run ID @ 13
Museum d'Histoire Naturelle Ville de Geneve @ 13
Morphosource @ 11
Bird Rescue Center, Santa Rosa @ 11
CHAS:Teach @ 10
LMNM: Landesmuseum Natur und Mensch, Oldenburg, Germany @ 10
California State University, Long Beach @ 10
Museo Nacional de Historia Natural, Santo Domingo @ 9
Original Image Number @ 9
PSM:Mamm Puget Sound Museum @ 9
San Diego Zoo @ 8
DZTH: Denver Zoology Tissue Herp @ 8
University of Minnesota-Duluth @ 7
UCLA: University of California Los Angeles @ 7
YPM: Yale Peabody Museum @ 7
CHAS:Herb @ 7
DGR:Bird @ 7
UCR: University of California Riverside Herbarium @ 7
Universidad de Costa Rica @ 7
MLZ:Bird @ 7
MLZ:Egg @ 7
UWBM: Genetic Resources Collection @ 7
UAM:Arc Locality ID @ 6
WNMU: Western New Mexico University @ 6
NMDGF: New Mexico Department of Game and Fish @ 6
MCZ:Ichthyology @ 6
La Sierra University Herpetology Collection @ 6
Southern Illinois University, Carbondale, Fluid Vertebrate Collection @ 6
UAM:Inv @ 6
UMNH:Mamm @ 6
MCZ:Mamm @ 5
ROM: Royal Ontario Museum Field Catalog Series @ 5
NTU: National Taiwan University, Taipei @ 5
CHAS:Herp @ 5
Navajo Nation Heritage Program @ 5
Stanford University Natural History Museum @ 5
Florida Wildlife Hospital @ 5
SDSU: San Diego State University @ 5
Museo de Vertebrados, Universidad de Panamá, Panamá City @ 4
Collection of Vertebrates, University of Texas, Arlington @ 4
Mpala Research Station @ 4
OWU:Fish @ 4
CAS: California Academy of Sciences (Stanford University), San Francisco @ 4
Fort Worth Zoo @ 4
Peregrine Fund @ 4
Louisiana State University, Museum of Natural Science @ 3
UWZM: University of Wisconsin Zoological Museum @ 3
MVZ: Museum of Vertebrate Zoology @ 3
Pinnacles National Monument @ 3
UAM:Bird @ 3
AB: Albuquerque Biopark @ 3
UAMObs:Mamm @ 3
toe clip number @ 3
OSUS: Oklahoma State University Collection of Vertebrates  @ 3
UAM:Arc @ 3
UAZ: University of Arizona @ 3
UCM:Herp @ 3
MSB:Herp @ 3
Musee National d'histoire Naturelle, Paris @ 3
Museo de Zoología, Universidad de Costa Rica, San José @ 2
Hebrew University of Jerusalem @ 2
Universidade Federal de Minas Gerais, Belahorizonte @ 2
UTEP:Herb @ 2
UWYMV Historia Catalog Vertebrate @ 2
UTEPObs:Herp @ 2
MVZ:Hild @ 2
UCB (ESPM): University of California Berkeley @ 2
Sea World, San Diego @ 2
Dominion Museum, Wellington, New Zealand @ 2
Albertson College Museum of Natural History @ 2
UCM:Obs @ 2
UFC: University of Florida Collections @ 2
UTEPObs:Ento @ 2
Chicago Natural History Museum (currently FMNH) @ 2
CSU-Chico: California State University, Chico @ 2
ALMNH:ES @ 2
Donald R. Dickey Collection @ 2
GBIF occurrence ID @ 2
UAM:ES @ 1
DMNS:Inv @ 1
CUMV:Mamm @ 1
Audubon Zoo @ 1
UPS: University of Puget Sound, Tacoma, Washington @ 1
Chiba Natural History Museum, Japan @ 1
Institut des Sciences de l'Evolution, Universite Montpellier II @ 1
UMNH:Bird @ 1
ARK @ 1
UTEP:Fish @ 1
UAMb:Herb @ 1
Los Angeles Zoo @ 1
TU:Fish @ 1
Texas Department of Fish and Game @ 1
KNWR:ENTO @ 1
Rotterdam Zoo @ 1
San Jose State University @ 1
Montana State University @ 1
U. S. Bureau of Biological Survey @ 1
CHAS:EH @ 1
UWYMV:Mamm @ 1
Universidad Nacional Autónoma de Honduras @ 1
Maryland Zoo @ 1
ANSP: Academy of Natural Sciences of Drexel University @ 1
LFL: CSU Larval Fish Lab @ 1

291 rows selected.

Elapsed: 00:00:00.68
UAM@ARCTOS> 
```

**Comment by Jegelewicz** [comment]
@campmlc if you select NK as your preferred other ID to make things easy on the search screen. I requested this because when I go to see if a museum code has been entered, just looking at the code table in search it is very difficult to find things because there is no apparent order.

As with the geology code table, perhaps we need a search when viewing the table. It is super difficult as an incoming collection to determine if an other ID I want to use is already in the table or not when you don't have access to edit the code table. You basically just wait until a bulkload fails or you load it as something generic.

As for this

> Is anyone searching on "LFL: CSU Larval Fish Lab" (or any of the seldom-used types which don't have a URL)? Would having that as type="original identifier" + value="LFL: CSU Larval Fish Lab Number Twelve" somehow reduce the quality of the data? If the answer is "not really" then why do we have it? http://handbook.arctosdb.org/documentation/other-identifying-numbers.html#general-guidelines

It is difficult to determine what collections will come in the future that may also want to use this seeminly obscure other ID. Someone asked for it for a reason and although it makes things difficult, I like being able to search LFL: CSU Larval Fish Lab Number Twelve across Arctos and know that I won't miss things that are original identifier = LFL: CSU Larval Fish Lab # Twelve.

**Comment by dustymc** [comment]
>As with the geology code table, perhaps we need a search when viewing the table.

Geology data are inherently searchable - anyone using it knows there's probably a 'Pleistocene' somewhere in there, even if they're not very familiar with Arctos. Nobody is going to guess that there's a "LFL: CSU Larval Fish Lab" in this thing. Making users guess at stuff we've made up is never good UI.

>don't have access to edit the code table

I'm not sure why they'd need that - code tables are public (and that's splat-F searchable), there should be (but probably aren't always) 'define' links by the selects, etc.

If you want to "search LFL: CSU Larval Fish Lab Number Twelve across Arctos" then you'll need some control of the values, which is best done by adding a base_URL and limiting them to things the external org has assigned. (We've already got two versions of #⁠12 in this comment - oops, there's number three - and that one contains a non-display character) What that DOES support is searching for the type. That obviously has some value, but at the cost of this list trending towards infinite length - and the longer the list gets the more likely someone won't notice our variation of Colorado State University Larval Fish Laboratory and will stuff their data into some generic ID.
